extensions/mssql/src/objectManagement/ui/databaseDialog.ts

```diff
@@ -431,6 +476,324 @@ export class DatabaseDialog extends ObjectManagementDialogBase<Database, Databas
+					values: this.viewInfo.dscEnableDisableOptions
+					, value: currentRowObjectInfo.valueForSecondary
+				});
+			}
+		}
+		// All other options accepts primary and seconday values as ON/OFF/PRIMARY(only secondary)
+		else {
+			await this.showDropdownsSection(isSecondaryCheckboxChecked);
+			if (JSON.stringify(this.valueForPrimaryDropdown.values) !== JSON.stringify(this.viewInfo.dscOnOffOptions) ||
+				this.valueForPrimaryDropdown.value !== currentRowObjectInfo.valueForPrimary) {
+				await this.valueForPrimaryDropdown.updateProperties({
+					values: this.viewInfo.dscOnOffOptions
+					, value: currentRowObjectInfo.valueForPrimary
+				});
+			}
+			if (JSON.stringify(this.valueForSecondaryDropdown.values) !== JSON.stringify(this.viewInfo.dscOnOffOptions) ||
+				this.valueForSecondaryDropdown.value !== currentRowObjectInfo.valueForSecondary) {
+				await this.valueForSecondaryDropdown.updateProperties({
+					values: this.viewInfo.dscOnOffOptions
+					, value: currentRowObjectInfo.valueForSecondary
+				});
+			}
+		}
+	}
+
+	/**
+	 * Initializes primary and secondary values of Input type
+	 * @returns GroupContainer for primary value and secondary value
+	 */
+	private async initializeDscValueInputTypeSection(): Promise<azdata.GroupContainer> {
+		// Primary value
+		this.valueForPrimaryInput = this.createInputBox(localizedConstants.ValueForPrimaryColumnHeader, async (newValue) => {
+			if (this.objectInfo.databaseScopedConfigurations[this.currentRowId].valueForPrimary !== newValue) {
+				this.objectInfo.databaseScopedConfigurations[this.currentRowId].valueForPrimary = newValue;
+				if (this.dscTable.data[this.currentRowId][1] !== newValue) {
+					this.dscTable.data[this.currentRowId][1] = newValue;
+				}
+				// Update the secondary value with the primary, when the set seconadry checkbox is checked
+				if (this.setSecondaryCheckboxForInputType.checked
+					&& this.objectInfo.databaseScopedConfigurations[this.currentRowId].id !== 25) {
+					this.objectInfo.databaseScopedConfigurations[this.currentRowId].valueForSecondary = newValue;
+					this.dscTable.data[this.currentRowId][2] = newValue;
+				}
+				await this.updateDscTable(this.dscTable.data, this.valueForPrimaryInput);
+			}
+		}, '', true, 'number', 150, false, 0, 0);
+		const primaryContainer = this.createLabelInputContainer(localizedConstants.ValueForPrimaryColumnHeader, this.valueForPrimaryInput);
+		this.dscPrimaryValueInputGroup = this.createGroup('', [primaryContainer], false, true);
+		await this.dscPrimaryValueInputGroup.updateCssStyles({ 'visibility': 'hidden' });
+
+		// Apply Primary To Secondary checkbox
+		this.setSecondaryCheckboxForInputType = this.createCheckbox(localizedConstants.SetSecondaryText, async (checked) => {
+			await this.dscSecondaryValueInputGroup.updateCssStyles({ 'visibility': checked ? 'hidden' : 'visible' });
+			this.objectInfo.databaseScopedConfigurations[this.currentRowId].valueForSecondary = checked
```

Comment:
Should this be an equality comparison here rather than an assignment? I.e. `===` vs `=`

Reply:
We are setting valueForSecondary here, if checked ? valueForPriamry : originalValueForSecondary! used a ternary condition in the next lines.